memory_model/src/guest_memory.rs

```diff
@@ -138,6 +146,112 @@ impl GuestMemory {
         self.regions.len()
     }
 
+    fn get_pfn(num: u64) -> u64 {
+        num & 0x7FFFFFFFFFFFFF
+    }
+
+    /// return a mapping from guest physical page numbers to host physical page numbers
+    /// for the given virtual address range
+    fn read_pagemap_addr_range(page_i_base: usize, page_size: u64, addr: u64, size: u64)
```

Comment:
the name `read_*` implies that it'll read the memory rather than translate. perhaps `get_pagemap_addr_range` instead?

Reply:
So all functions added in this file except for `dump init` and `load_init` are for idle page tracking. Do you think it's better to remove all dirty or idle page tracking functions (ones here and ones added in `vmm/src/vstate.rs`) for this pull request?

Reply:
I do think the pull request is better off if it's limited to a particular feature, but don't we need dirty page tracking in order to snapshot the minimal amount of memory?

Reply:
Right.. I completely forgot about this part. Then we need to keep the ones in `memory_model/src/guest_memory.rs` since they are used by `dump_init` but we can remove all page tracking functions in `vmm/src/vstate.rs` those are used to do memory analysis.